torch/fx/graph.py

```diff
@@ -66,9 +67,17 @@ def map_arg(a: Argument, fn: Callable[[Node], Argument]) -> Argument:
         return a
 
 class Graph:
-    def __init__(self):
+    def __init__(self, annotations_dict : Optional[Dict[str, Any]] = None):
+        """
+        Construct an empty Graph. `annotations_dict` is an optional argument that
+        takes the dict that would appear on `__annotations__` for the Python
+        function that corresponds to this Graph. The annotations will be
+        used during code generation to ensure that arguments and returns have
+        the correct type annotations.
+        """
         self._nodes : List[Node] = []
         self._used_names : Dict[str, int] = {}  # base name -> number
+        self.annotations = annotations_dict if annotations_dict else {}
```

Comment:
why is it attached on the graph level and not a field on Node level? Arguments for the latter:
* easier to preserve types if I remove/rename the inputs (as the annotation lives on the placeholder node directly)
* may be convenient to preserve annotations on some intermediate nodes, e.g. if I happen to inline and un-inline some method and want to restore the annotations
* I'd argue that we might even want to capture output annotations when some module is called (just in case)

Reply:
Originally I was just going to attach this dict as `__annotations__` on the generated `forward()` function, but it turns out that doesn't work so I ended up generating the type annotations as code. Agree that after that change this API doesn't make much sense (for my reasons and your reasons)